cpp/include/cudf/detail/aggregation/aggregation.cuh

```diff
@@ -0,0 +1,233 @@
+      return;
+    }
+
+    using Target = target_type_t<Source, aggregation::ARGMIN>;
+    auto old = atomicCAS(&target.element<Target>(target_index), ARGMIN_SENTINEL,
+                         source_index);
+    if (old == ARGMIN_SENTINEL) {
+      return;
+    }
+
+    while (source.element<Source>(source_index) < source.element<Source>(old)) {
+      old = atomicCAS(&target.element<Target>(target_index), old, source_index);
+    }
+
+    if (target_has_nulls and target.is_null(target_index)) {
+      target.set_valid(target_index);
+    }
+  }
+};
+
+/**
+ * @brief Function object to update a single element in a target column by
+ * performing an aggregation operation with a single element from a source
+ * column.
+ *
+ * @tparam target_has_nulls Indicates presence of null elements in `target`
+ * @tparam source_has_nulls Indicates presence of null elements in `source`.
+ */
+template <bool target_has_nulls = true, bool source_has_nulls = true>
+struct elementwise_aggregator {
+  template <typename Source, aggregation::Kind k>
+  __device__ void operator()(mutable_column_device_view target,
+                             size_type target_index, column_device_view source,
+                             size_type source_index) const noexcept {
+    update_target_element<Source, k, target_has_nulls, source_has_nulls>{}(
+        target, target_index, source, source_index);
+  }
+};
+
+/**
+ * @brief Updates a row in `target` by performing elementwise aggregation
+ * operations with a row in `source`.
+ *
+ * For the row in `target` specified by `target_index`, each element at `i` is
+ * updated by:
+ * ```c++
+ * target_row[i] = aggs[i](target_row[i], source_row[i])
+ * ```
+ * TODO: Document null handling, expectations for initialization/validity of output columns
+ *
+ * @param target Table containing the row to update
+ * @param target_index Index of the row to update in `target`
+ * @param source Table containing the row used to update the row in `target`.
+ * The invariant `source.num_columns() >= target.num_columns()` must hold.
+ * @param source_index Index of the row to use in `source`
+ * @param aggs Array of aggregations to perform between elements of the `target`
+ * and `source` rows. Must contain at least `target.num_columns()` valid
+ * `aggregation::Kind` values.
+ */
+// template <bool target_has_nulls = true, bool source_has_nulls = true>
+__device__ inline void aggregate_row(mutable_table_device_view target,
+                                     size_type target_index,
+                                     table_device_view source,
+                                     size_type source_index,
+                                     aggregation::Kind* aggs) {
+  for (auto i = 0; i < target.num_columns(); ++i) {
+    dispatch_type_and_aggregation(
+        target.column(i).type(), aggs[i], elementwise_aggregator<true, true>{},
+        // elementwise_aggregator<target_has_nulls, source_has_nulls>{},
```

Comment:
Is this ready to be enabled?

Reply:
Yeah, I just explicitly specified the template arguments to force compilation. Otherwise, since these functions aren't actually used anywhere, they will never get compiled.